chpl/chpl-service/src/main/java/gov/healthit/chpl/attestation/manager/AttestationManager.java

```diff
@@ -57,4 +74,92 @@ public DeveloperAttestationSubmission saveDeveloperAttestation(DeveloperAttestat
     public List<DeveloperAttestationSubmission> getDeveloperAttestations(Long developerId) {
         return attestationDAO.getDeveloperAttestationSubmissionsByDeveloper(developerId);
     }
+
+    @Transactional
+    public Boolean canDeveloperSubmitChangeRequest(Long developerId) throws EntityRetrievalException {
+        AttestationPeriod mostRecentPastAttestationPeriod = attestationPeriodService.getMostRecentPastAttestationPeriod();
+        if (doesAttestationForDeveloperExist(developerId, mostRecentPastAttestationPeriod.getId())) {
+            if (doesPendingAttestationChangeRequestForDeveloperExist(developerId)) {
+                return false;
+            } else {
+                return isDateInFuture(attestationPeriodService.getMostRecentPeriodExceptionDateForDeveloper(developerId));
+            }
+        } else {
+            return attestationPeriodService.isDateWithinSubmissionPeriodForDeveloper(developerId, LocalDate.now());
+        }
+    }
+
+    @Transactional
+    public Boolean canCreateException(Long developerId) throws EntityRetrievalException {
+        if (canDeveloperSubmitChangeRequest(developerId)) {
+            return false;
+        } else {
+            AttestationPeriod mostRecentPastAttestationPeriod = attestationPeriodService.getMostRecentPastAttestationPeriod();
+            if (withinStandardSubmissionPeriod(mostRecentPastAttestationPeriod)) {
+                if (doesAttestationForDeveloperExist(developerId, mostRecentPastAttestationPeriod.getId())) {
+                    return !doesPendingAttestationChangeRequestForDeveloperExist(developerId);
+                } else {
+                    return false;
+                }
+            } else {
+                return !doesPendingAttestationChangeRequestForDeveloperExist(developerId);
```

Comment:
I feel like the "withinStandardsSubmissionPeriod" part is unnecessary.

If we're in this block the Developer can't submit one, so the only reason an admin couldn't create an exception is if the Developer couldn't submit one due to a pending CR. I don't think there's anything in that logic that depends on the "standards submission period"

I feel like I need a flowchart on this, but would like to investigate the states that get us to each of these branches, as it'd be nice to prune out paths that we might not actually need

Reply:
I think we do need the `withinStandardSubmissionPeriod` check...  

In the standard submission window, an ACB should only be able to create an exception when an attestation exists for the developer.  Otherwise, if the ACB were to create an exception, before the attestation for the developer has been accepted, it could make the submission window period shorter than the standard submission window when an attestation does not exist.

I know this is hard to explain and maybe we should discuss.

Reply:
I agree with that logic, but on line 95 we check to see if a Developer can submit an attestation, and if so, return false from this method, never getting to this point. So if we're in the standard window, then a Developer can already create one, unless one is pending, or one already is published. In the first case the ACB can't create an attestation, which this line checks). In the second the ACB can, which this line also verifies. No logic depends on the standard window for either of those checks